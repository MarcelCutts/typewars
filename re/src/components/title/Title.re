module Styles = {
  open Css;

  let container =
    style([
      color(white),
      padding(px(20)),
      background(rgb(219, 77, 63)),
      fontFamily("montserrat"),
    ]);
};

let component = ReasonReact.statelessComponent("Title");

let make = (~content, _children) => {
  ...component,
  render: _self =>
    <h1 className=Styles.container> (content |> ReasonReact.string) </h1>,
};