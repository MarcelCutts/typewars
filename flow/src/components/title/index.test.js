// @flow strict
import React from "react";
import { shallow } from "enzyme";
import Title from ".";
import { renderToStringWithData } from "react-apollo";

describe("Title component", () => {
  test("renders", () => {
    const component = shallow(<Title content="test message" />);
    expect(component).toMatchSnapshot();
  });

  test("renders content as heading", () => {
    const component = shallow(<Title content="test" />);
    expect(component.childAt(0).text()).toEqual("test");
  });
});
