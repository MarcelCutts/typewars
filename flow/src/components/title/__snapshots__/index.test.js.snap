// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Title component renders 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Title
    content="test message"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": "test message",
    },
    "ref": null,
    "rendered": "test message",
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": "test message",
      },
      "ref": null,
      "rendered": "test message",
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
